diff -ruN a/src/BotApi.php b/src/BotApi.php
--- a/src/BotApi.php	2026-05-02 21:04:04.960136765 +0300
+++ b/src/BotApi.php	2026-05-02 21:04:14.627186783 +0300
@@ -70,7 +70,7 @@
      *
      * @return mixed
      */
-    public function call(MethodInterface $method, string $type = null)
+    public function call(MethodInterface $method, ?string $type = null)
     {
         $json = $this->apiClient->send($this->getMethodName($method), $this->normalizer->normalize($method));
 
diff -ruN a/src/BotApiInterface.php b/src/BotApiInterface.php
--- a/src/BotApiInterface.php	2026-05-02 21:04:04.931076627 +0300
+++ b/src/BotApiInterface.php	2026-05-02 21:04:14.591799444 +0300
@@ -44,7 +44,7 @@
      *
      * @return mixed
      */
-    public function call(MethodInterface $method, string $type = null);
+    public function call(MethodInterface $method, ?string $type = null);
 
     /**
      * @param ExportChatInviteLinkMethod $method
diff -ruN a/src/Method/AddStickerToSetMethod.php b/src/Method/AddStickerToSetMethod.php
--- a/src/Method/AddStickerToSetMethod.php	2026-05-02 21:04:04.966794834 +0300
+++ b/src/Method/AddStickerToSetMethod.php	2026-05-02 21:04:14.633274238 +0300
@@ -73,7 +73,7 @@
         string $name,
         $pngSticker,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): AddStickerToSetMethod {
         return static::createStatic($userId, $name, $pngSticker, $emojis, $data);
     }
@@ -88,7 +88,7 @@
         string $name,
         $pngSticker,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): AddStickerToSetMethod {
         $instance = static::createBase($userId, $name, $emojis, $data);
         $instance->pngSticker = $pngSticker;
@@ -104,7 +104,7 @@
         string $name,
         InputFileType $tgsSticker,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): AddStickerToSetMethod {
         $instance = static::createBase($userId, $name, $emojis, $data);
         $instance->tgsSticker = $tgsSticker;
@@ -119,7 +119,7 @@
         int $userId,
         string $name,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): AddStickerToSetMethod {
         $instance = new static();
         $instance->userId = $userId;
diff -ruN a/src/Method/AnswerCallbackQueryMethod.php b/src/Method/AnswerCallbackQueryMethod.php
--- a/src/Method/AnswerCallbackQueryMethod.php	2026-05-02 21:04:04.979617189 +0300
+++ b/src/Method/AnswerCallbackQueryMethod.php	2026-05-02 21:04:14.648779661 +0300
@@ -69,7 +69,7 @@
      *
      * @return AnswerCallbackQueryMethod
      */
-    public static function create(string $callbackQueryId, array $data = null): AnswerCallbackQueryMethod
+    public static function create(string $callbackQueryId, ?array $data = null): AnswerCallbackQueryMethod
     {
         $instance = new static();
         $instance->callbackQueryId = $callbackQueryId;
diff -ruN a/src/Method/AnswerInlineQueryMethod.php b/src/Method/AnswerInlineQueryMethod.php
--- a/src/Method/AnswerInlineQueryMethod.php	2026-05-02 21:04:04.975095450 +0300
+++ b/src/Method/AnswerInlineQueryMethod.php	2026-05-02 21:04:14.642828787 +0300
@@ -91,7 +91,7 @@
      *
      * @return AnswerInlineQueryMethod
      */
-    public static function create(string $inlineQueryId, array $results, array $data = null): AnswerInlineQueryMethod
+    public static function create(string $inlineQueryId, array $results, ?array $data = null): AnswerInlineQueryMethod
     {
         $instance = new static();
         $instance->inlineQueryId = $inlineQueryId;
diff -ruN a/src/Method/CopyMessageMethod.php b/src/Method/CopyMessageMethod.php
--- a/src/Method/CopyMessageMethod.php	2026-05-02 21:04:04.967679190 +0300
+++ b/src/Method/CopyMessageMethod.php	2026-05-02 21:04:14.634449963 +0300
@@ -41,7 +41,7 @@
      * @param $chatId
      * @param $fromChatId
      */
-    public static function create($chatId, $fromChatId, int $messageId, array $data = null): CopyMessageMethod
+    public static function create($chatId, $fromChatId, int $messageId, ?array $data = null): CopyMessageMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/CreateNewStickerSetMethod.php b/src/Method/CreateNewStickerSetMethod.php
--- a/src/Method/CreateNewStickerSetMethod.php	2026-05-02 21:04:04.982168177 +0300
+++ b/src/Method/CreateNewStickerSetMethod.php	2026-05-02 21:04:14.651657684 +0300
@@ -93,7 +93,7 @@
         string $title,
         $pngSticker,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): CreateNewStickerSetMethod {
         return static::createStatic($userId, $name, $title, $pngSticker, $emojis, $data);
     }
@@ -111,7 +111,7 @@
         string $title,
         $pngSticker,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): CreateNewStickerSetMethod {
         $instance = static::createBase($userId, $name, $title, $emojis, $data);
         $instance->pngSticker = $pngSticker;
@@ -130,7 +130,7 @@
         string $title,
         InputFileType $tgsSticker,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): CreateNewStickerSetMethod {
         $instance = static::createBase($userId, $name, $title, $emojis, $data);
         $instance->tgsSticker = $tgsSticker;
@@ -146,7 +146,7 @@
         string $name,
         string $title,
         string $emojis,
-        array $data = null
+        ?array $data = null
     ): CreateNewStickerSetMethod {
         $instance = new static();
         $instance->userId = $userId;
diff -ruN a/src/Method/DeleteWebhookMethod.php b/src/Method/DeleteWebhookMethod.php
--- a/src/Method/DeleteWebhookMethod.php	2026-05-02 21:04:04.980929661 +0300
+++ b/src/Method/DeleteWebhookMethod.php	2026-05-02 21:04:14.650365670 +0300
@@ -26,7 +26,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(array $data = null): DeleteWebhookMethod
+    public static function create(?array $data = null): DeleteWebhookMethod
     {
         $instance = new static();
 
diff -ruN a/src/Method/EditMessageCaptionMethod.php b/src/Method/EditMessageCaptionMethod.php
--- a/src/Method/EditMessageCaptionMethod.php	2026-05-02 21:04:04.983493190 +0300
+++ b/src/Method/EditMessageCaptionMethod.php	2026-05-02 21:04:14.653601435 +0300
@@ -28,7 +28,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function create($chatId, int $messageId, array $data = null): EditMessageCaptionMethod
+    public static function create($chatId, int $messageId, ?array $data = null): EditMessageCaptionMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
@@ -43,7 +43,7 @@
     /**
      * @throws BadArgumentException
      */
-    public static function createInline(string $inlineMessageId, array $data = null): EditMessageCaptionMethod
+    public static function createInline(string $inlineMessageId, ?array $data = null): EditMessageCaptionMethod
     {
         $instance = new static();
         $instance->inlineMessageId = $inlineMessageId;
diff -ruN a/src/Method/EditMessageLiveLocationMethod.php b/src/Method/EditMessageLiveLocationMethod.php
--- a/src/Method/EditMessageLiveLocationMethod.php	2026-05-02 21:04:04.981890891 +0300
+++ b/src/Method/EditMessageLiveLocationMethod.php	2026-05-02 21:04:14.651294442 +0300
@@ -66,7 +66,7 @@
         int $messageId,
         float $latitude,
         float $longitude,
-        array $data = null
+        ?array $data = null
     ): EditMessageLiveLocationMethod {
         $instance = new static();
         $instance->chatId = $chatId;
@@ -87,7 +87,7 @@
         string $inlineMessageId,
         float $latitude,
         float $longitude,
-        array $data = null
+        ?array $data = null
     ): EditMessageLiveLocationMethod {
         $instance = new static();
         $instance->latitude = $latitude;
diff -ruN a/src/Method/EditMessageMediaMethod.php b/src/Method/EditMessageMediaMethod.php
--- a/src/Method/EditMessageMediaMethod.php	2026-05-02 21:04:04.984373380 +0300
+++ b/src/Method/EditMessageMediaMethod.php	2026-05-02 21:04:14.654618372 +0300
@@ -40,7 +40,7 @@
         $chatId,
         int $messageId,
         InputMediaType $media,
-        array $data = null
+        ?array $data = null
     ): EditMessageMediaMethod {
         $instance = new static();
         $instance->chatId = $chatId;
@@ -65,7 +65,7 @@
     public static function createInline(
         string $inlineMessageId,
         InputMediaType $media,
-        array $data = null
+        ?array $data = null
     ): EditMessageMediaMethod {
         $instance = new static();
         $instance->inlineMessageId = $inlineMessageId;
diff -ruN a/src/Method/EditMessageReplyMarkupMethod.php b/src/Method/EditMessageReplyMarkupMethod.php
--- a/src/Method/EditMessageReplyMarkupMethod.php	2026-05-02 21:04:04.967448695 +0300
+++ b/src/Method/EditMessageReplyMarkupMethod.php	2026-05-02 21:04:14.634135511 +0300
@@ -27,7 +27,7 @@
      *
      * @return EditMessageReplyMarkupMethod
      */
-    public static function create($chatId, int $messageId, array $data = null): EditMessageReplyMarkupMethod
+    public static function create($chatId, int $messageId, ?array $data = null): EditMessageReplyMarkupMethod
     {
         $instance = new self();
         $instance->chatId = $chatId;
@@ -49,7 +49,7 @@
      */
     public static function createInline(
         string $inlineMessageId,
-        array $data = null
+        ?array $data = null
     ): EditMessageReplyMarkupMethod {
         $instance = new self();
         $instance->inlineMessageId = $inlineMessageId;
diff -ruN a/src/Method/EditMessageTextMethod.php b/src/Method/EditMessageTextMethod.php
--- a/src/Method/EditMessageTextMethod.php	2026-05-02 21:04:04.982610251 +0300
+++ b/src/Method/EditMessageTextMethod.php	2026-05-02 21:04:14.652372336 +0300
@@ -54,7 +54,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, int $messageId, string $text, array $data = null): EditMessageTextMethod
+    public static function create($chatId, int $messageId, string $text, ?array $data = null): EditMessageTextMethod
     {
         $instance = new self();
         $instance->chatId = $chatId;
@@ -73,7 +73,7 @@
     public static function createInline(
         string $inlineMessageId,
         string $text,
-        array $data = null
+        ?array $data = null
     ): EditMessageTextMethod {
         $instance = new self();
         $instance->inlineMessageId = $inlineMessageId;
diff -ruN a/src/Method/ForwardMessageMethod.php b/src/Method/ForwardMessageMethod.php
--- a/src/Method/ForwardMessageMethod.php	2026-05-02 21:04:04.970335134 +0300
+++ b/src/Method/ForwardMessageMethod.php	2026-05-02 21:04:14.637363943 +0300
@@ -51,7 +51,7 @@
      *
      * @return ForwardMessageMethod
      */
-    public static function create($chatId, $fromChatId, int $messageId, array $data = null): self
+    public static function create($chatId, $fromChatId, int $messageId, ?array $data = null): self
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/GetChatMenuButtonMethod.php b/src/Method/GetChatMenuButtonMethod.php
--- a/src/Method/GetChatMenuButtonMethod.php	2026-05-02 21:04:04.967884894 +0300
+++ b/src/Method/GetChatMenuButtonMethod.php	2026-05-02 21:04:14.634830580 +0300
@@ -35,7 +35,7 @@
      *
      * @return GetChatMenuButtonMethod
      */
-    public static function create(array $data = null): GetChatMenuButtonMethod
+    public static function create(?array $data = null): GetChatMenuButtonMethod
     {
         $instance = new static();
         if ($data) {
diff -ruN a/src/Method/GetUpdatesMethod.php b/src/Method/GetUpdatesMethod.php
--- a/src/Method/GetUpdatesMethod.php	2026-05-02 21:04:04.984639999 +0300
+++ b/src/Method/GetUpdatesMethod.php	2026-05-02 21:04:14.654980989 +0300
@@ -66,7 +66,7 @@
      *
      * @return GetUpdatesMethod
      */
-    public static function create(array $data = null): GetUpdatesMethod
+    public static function create(?array $data = null): GetUpdatesMethod
     {
         $instance = new static();
         if ($data) {
diff -ruN a/src/Method/GetUserProfilePhotosMethod.php b/src/Method/GetUserProfilePhotosMethod.php
--- a/src/Method/GetUserProfilePhotosMethod.php	2026-05-02 21:04:04.981220738 +0300
+++ b/src/Method/GetUserProfilePhotosMethod.php	2026-05-02 21:04:14.650706663 +0300
@@ -42,7 +42,7 @@
      *
      * @return GetUserProfilePhotosMethod
      */
-    public static function create(int $userId, array $data = null): GetUserProfilePhotosMethod
+    public static function create(int $userId, ?array $data = null): GetUserProfilePhotosMethod
     {
         $instance = new static();
         $instance->userId = $userId;
diff -ruN a/src/Method/KickChatMemberMethod.php b/src/Method/KickChatMemberMethod.php
--- a/src/Method/KickChatMemberMethod.php	2026-05-02 21:04:04.978506045 +0300
+++ b/src/Method/KickChatMemberMethod.php	2026-05-02 21:04:14.647587561 +0300
@@ -39,7 +39,7 @@
      *
      * @return KickChatMemberMethod
      */
-    public static function create($chatId, int $userId, array $data = null): KickChatMemberMethod
+    public static function create($chatId, int $userId, ?array $data = null): KickChatMemberMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/PinChatMessageMethod.php b/src/Method/PinChatMessageMethod.php
--- a/src/Method/PinChatMessageMethod.php	2026-05-02 21:04:04.974466964 +0300
+++ b/src/Method/PinChatMessageMethod.php	2026-05-02 21:04:14.641721977 +0300
@@ -37,7 +37,7 @@
      *
      * @return PinChatMessageMethod
      */
-    public static function create($chatId, int $messageId, array $data = null): PinChatMessageMethod
+    public static function create($chatId, int $messageId, ?array $data = null): PinChatMessageMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/PromoteChatMemberMethod.php b/src/Method/PromoteChatMemberMethod.php
--- a/src/Method/PromoteChatMemberMethod.php	2026-05-02 21:04:04.978040514 +0300
+++ b/src/Method/PromoteChatMemberMethod.php	2026-05-02 21:04:14.646904409 +0300
@@ -91,7 +91,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, $userId, array $data = null): PromoteChatMemberMethod
+    public static function create($chatId, $userId, ?array $data = null): PromoteChatMemberMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/RestrictChatMemberMethod.php b/src/Method/RestrictChatMemberMethod.php
--- a/src/Method/RestrictChatMemberMethod.php	2026-05-02 21:04:04.982398963 +0300
+++ b/src/Method/RestrictChatMemberMethod.php	2026-05-02 21:04:14.652079509 +0300
@@ -87,7 +87,7 @@
      * @deprecated
      * @see https://core.telegram.org/bots/api#july-29-2019
      */
-    public static function createOld($chatId, int $userId, array $data = null): RestrictChatMemberMethod
+    public static function createOld($chatId, int $userId, ?array $data = null): RestrictChatMemberMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
@@ -113,7 +113,7 @@
         $chatId,
         int $userId,
         ChatPermissionsType $chatPermissionsType,
-        array $data = null
+        ?array $data = null
     ): RestrictChatMemberMethod {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendAnimationMethod.php b/src/Method/SendAnimationMethod.php
--- a/src/Method/SendAnimationMethod.php	2026-05-02 21:04:04.974662251 +0300
+++ b/src/Method/SendAnimationMethod.php	2026-05-02 21:04:14.641987055 +0300
@@ -72,7 +72,7 @@
      *
      * @return SendAnimationMethod
      */
-    public static function create($chatId, $animation, array $data = null): SendAnimationMethod
+    public static function create($chatId, $animation, ?array $data = null): SendAnimationMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendAudioMethod.php b/src/Method/SendAudioMethod.php
--- a/src/Method/SendAudioMethod.php	2026-05-02 21:04:04.969020412 +0300
+++ b/src/Method/SendAudioMethod.php	2026-05-02 21:04:14.635950098 +0300
@@ -71,7 +71,7 @@
      *
      * @return SendAudioMethod
      */
-    public static function create($chatId, $audio, array $data = null): SendAudioMethod
+    public static function create($chatId, $audio, ?array $data = null): SendAudioMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendContactMethod.php b/src/Method/SendContactMethod.php
--- a/src/Method/SendContactMethod.php	2026-05-02 21:04:04.976877746 +0300
+++ b/src/Method/SendContactMethod.php	2026-05-02 21:04:14.645236736 +0300
@@ -60,7 +60,7 @@
         $chatId,
         string $phoneNumber,
         string $firstName,
-        array $data = null
+        ?array $data = null
     ): SendContactMethod {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendDiceMethod.php b/src/Method/SendDiceMethod.php
--- a/src/Method/SendDiceMethod.php	2026-05-02 21:04:04.976652084 +0300
+++ b/src/Method/SendDiceMethod.php	2026-05-02 21:04:14.644979283 +0300
@@ -46,7 +46,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function create($chatId, array $data = null): SendDiceMethod
+    public static function create($chatId, ?array $data = null): SendDiceMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
@@ -63,7 +63,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function createWithDice($chatId, array $data = null): SendDiceMethod
+    public static function createWithDice($chatId, ?array $data = null): SendDiceMethod
     {
         $instance = static::create($chatId, $data);
         $instance->emoji = static::EMOJI_DICE;
@@ -76,7 +76,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function createWithDarts($chatId, array $data = null): SendDiceMethod
+    public static function createWithDarts($chatId, ?array $data = null): SendDiceMethod
     {
         $instance = static::create($chatId, $data);
         $instance->emoji = self::EMOJI_DARTS;
@@ -89,7 +89,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function createWithBasketball($chatId, array $data = null): SendDiceMethod
+    public static function createWithBasketball($chatId, ?array $data = null): SendDiceMethod
     {
         $instance = static::create($chatId, $data);
         $instance->emoji = self::EMOJI_BASKETBALL;
@@ -102,7 +102,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function createWithFootBall($chatId, array $data = null): SendDiceMethod
+    public static function createWithFootBall($chatId, ?array $data = null): SendDiceMethod
     {
         $instance = static::create($chatId, $data);
         $instance->emoji = self::EMOJI_FOOTBALL;
@@ -115,7 +115,7 @@
      *
      * @throws BadArgumentException
      */
-    public static function createWithSlotMachine($chatId, array $data = null): SendDiceMethod
+    public static function createWithSlotMachine($chatId, ?array $data = null): SendDiceMethod
     {
         $instance = static::create($chatId, $data);
         $instance->emoji = self::EMOJI_SLOT_MACHINE;
diff -ruN a/src/Method/SendDocumentMethod.php b/src/Method/SendDocumentMethod.php
--- a/src/Method/SendDocumentMethod.php	2026-05-02 21:04:04.979124491 +0300
+++ b/src/Method/SendDocumentMethod.php	2026-05-02 21:04:14.648197299 +0300
@@ -55,7 +55,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, $document, array $data = null): SendDocumentMethod
+    public static function create($chatId, $document, ?array $data = null): SendDocumentMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendGameMethod.php b/src/Method/SendGameMethod.php
--- a/src/Method/SendGameMethod.php	2026-05-02 21:04:04.968167221 +0300
+++ b/src/Method/SendGameMethod.php	2026-05-02 21:04:14.635091575 +0300
@@ -37,7 +37,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(int $chatId, string $gameShortName, array $data = null): SendGameMethod
+    public static function create(int $chatId, string $gameShortName, ?array $data = null): SendGameMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendInvoiceMethod.php b/src/Method/SendInvoiceMethod.php
--- a/src/Method/SendInvoiceMethod.php	2026-05-02 21:04:04.983081241 +0300
+++ b/src/Method/SendInvoiceMethod.php	2026-05-02 21:04:14.652979865 +0300
@@ -181,7 +181,7 @@
         string $startParameter,
         string $currency,
         array $prices,
-        array $data = null
+        ?array $data = null
     ): SendInvoiceMethod {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendLocationMethod.php b/src/Method/SendLocationMethod.php
--- a/src/Method/SendLocationMethod.php	2026-05-02 21:04:04.977102867 +0300
+++ b/src/Method/SendLocationMethod.php	2026-05-02 21:04:14.645539313 +0300
@@ -68,7 +68,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, float $latitude, float $longitude, array $data = null): SendLocationMethod
+    public static function create($chatId, float $latitude, float $longitude, ?array $data = null): SendLocationMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendMediaGroupMethod.php b/src/Method/SendMediaGroupMethod.php
--- a/src/Method/SendMediaGroupMethod.php	2026-05-02 21:04:04.966599254 +0300
+++ b/src/Method/SendMediaGroupMethod.php	2026-05-02 21:04:14.632941870 +0300
@@ -56,7 +56,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, array $media, array $data = null): SendMediaGroupMethod
+    public static function create($chatId, array $media, ?array $data = null): SendMediaGroupMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendMessageMethod.php b/src/Method/SendMessageMethod.php
--- a/src/Method/SendMessageMethod.php	2026-05-02 21:04:04.965816979 +0300
+++ b/src/Method/SendMessageMethod.php	2026-05-02 21:04:14.632082180 +0300
@@ -54,7 +54,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, string $text, array $data = null): SendMessageMethod
+    public static function create($chatId, string $text, ?array $data = null): SendMessageMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendPhotoMethod.php b/src/Method/SendPhotoMethod.php
--- a/src/Method/SendPhotoMethod.php	2026-05-02 21:04:04.985243445 +0300
+++ b/src/Method/SendPhotoMethod.php	2026-05-02 21:04:14.655728057 +0300
@@ -39,7 +39,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, $photo, array $data = null): SendPhotoMethod
+    public static function create($chatId, $photo, ?array $data = null): SendPhotoMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendPollMethod.php b/src/Method/SendPollMethod.php
--- a/src/Method/SendPollMethod.php	2026-05-02 21:04:04.975299946 +0300
+++ b/src/Method/SendPollMethod.php	2026-05-02 21:04:14.643326818 +0300
@@ -118,7 +118,7 @@
      *
      * @return SendPollMethod
      */
-    public static function create(string $chatId, string $question, array $options, array $data = null): self
+    public static function create(string $chatId, string $question, array $options, ?array $data = null): self
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendStickerMethod.php b/src/Method/SendStickerMethod.php
--- a/src/Method/SendStickerMethod.php	2026-05-02 21:04:04.980697958 +0300
+++ b/src/Method/SendStickerMethod.php	2026-05-02 21:04:14.650109133 +0300
@@ -38,7 +38,7 @@
      *
      * @return SendStickerMethod
      */
-    public static function create($chatId, $sticker, array $data = null): SendStickerMethod
+    public static function create($chatId, $sticker, ?array $data = null): SendStickerMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendVenueMethod.php b/src/Method/SendVenueMethod.php
--- a/src/Method/SendVenueMethod.php	2026-05-02 21:04:04.973549483 +0300
+++ b/src/Method/SendVenueMethod.php	2026-05-02 21:04:14.640730581 +0300
@@ -78,7 +78,7 @@
         float $longitude,
         string $title,
         string $address,
-        array $data = null
+        ?array $data = null
     ): SendVenueMethod {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendVideoMethod.php b/src/Method/SendVideoMethod.php
--- a/src/Method/SendVideoMethod.php	2026-05-02 21:04:04.977324404 +0300
+++ b/src/Method/SendVideoMethod.php	2026-05-02 21:04:14.645854015 +0300
@@ -79,7 +79,7 @@
      *
      * @return SendVideoMethod
      */
-    public static function create($chatId, $video, array $data = null): SendVideoMethod
+    public static function create($chatId, $video, ?array $data = null): SendVideoMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendVideoNoteMethod.php b/src/Method/SendVideoNoteMethod.php
--- a/src/Method/SendVideoNoteMethod.php	2026-05-02 21:04:04.974860122 +0300
+++ b/src/Method/SendVideoNoteMethod.php	2026-05-02 21:04:14.642316464 +0300
@@ -63,7 +63,7 @@
      *
      * @return SendVideoNoteMethod
      */
-    public static function create($chatId, $videoNote, array $data = null): SendVideoNoteMethod
+    public static function create($chatId, $videoNote, ?array $data = null): SendVideoNoteMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SendVoiceMethod.php b/src/Method/SendVoiceMethod.php
--- a/src/Method/SendVoiceMethod.php	2026-05-02 21:04:04.978870746 +0300
+++ b/src/Method/SendVoiceMethod.php	2026-05-02 21:04:14.647890139 +0300
@@ -47,7 +47,7 @@
      *
      * @return SendVoiceMethod
      */
-    public static function create($chatId, $voice, array $data = null): SendVoiceMethod
+    public static function create($chatId, $voice, ?array $data = null): SendVoiceMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SetChatDescriptionMethod.php b/src/Method/SetChatDescriptionMethod.php
--- a/src/Method/SetChatDescriptionMethod.php	2026-05-02 21:04:04.975504858 +0300
+++ b/src/Method/SetChatDescriptionMethod.php	2026-05-02 21:04:14.643608854 +0300
@@ -33,7 +33,7 @@
      *
      * @return SetChatDescriptionMethod
      */
-    public static function create($chatId, array $data = null): SetChatDescriptionMethod
+    public static function create($chatId, ?array $data = null): SetChatDescriptionMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/SetChatMenuButtonMethod.php b/src/Method/SetChatMenuButtonMethod.php
--- a/src/Method/SetChatMenuButtonMethod.php	2026-05-02 21:04:04.971126326 +0300
+++ b/src/Method/SetChatMenuButtonMethod.php	2026-05-02 21:04:14.638119178 +0300
@@ -46,7 +46,7 @@
      *
      * @return SetChatMenuButtonMethod
      */
-    public static function create(array $data = null): SetChatMenuButtonMethod
+    public static function create(?array $data = null): SetChatMenuButtonMethod
     {
         $instance = new static();
         if ($data) {
diff -ruN a/src/Method/SetGameScoreMethod.php b/src/Method/SetGameScoreMethod.php
--- a/src/Method/SetGameScoreMethod.php	2026-05-02 21:04:04.977816935 +0300
+++ b/src/Method/SetGameScoreMethod.php	2026-05-02 21:04:14.646571208 +0300
@@ -82,7 +82,7 @@
         int $score,
         int $chatId,
         int $messageId,
-        array $data = null
+        ?array $data = null
     ): SetGameScoreMethod {
         $instance = new static();
         $instance->userId = $userId;
@@ -110,7 +110,7 @@
         int $userId,
         int $score,
         string $inlineMessageId,
-        array $data = null
+        ?array $data = null
     ): SetGameScoreMethod {
         $instance = new static();
         $instance->userId = $userId;
diff -ruN a/src/Method/SetWebhookMethod.php b/src/Method/SetWebhookMethod.php
--- a/src/Method/SetWebhookMethod.php	2026-05-02 21:04:04.973738812 +0300
+++ b/src/Method/SetWebhookMethod.php	2026-05-02 21:04:14.641032783 +0300
@@ -75,7 +75,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(string $url, array $data = null): SetWebhookMethod
+    public static function create(string $url, ?array $data = null): SetWebhookMethod
     {
         $instance = new static();
         $instance->url = $url;
diff -ruN a/src/Method/StopMessageLiveLocationMethod.php b/src/Method/StopMessageLiveLocationMethod.php
--- a/src/Method/StopMessageLiveLocationMethod.php	2026-05-02 21:04:04.970119722 +0300
+++ b/src/Method/StopMessageLiveLocationMethod.php	2026-05-02 21:04:14.637057033 +0300
@@ -27,7 +27,7 @@
      *
      * @return StopMessageLiveLocationMethod
      */
-    public static function create($chatId, int $messageId, array $data = null): StopMessageLiveLocationMethod
+    public static function create($chatId, int $messageId, ?array $data = null): StopMessageLiveLocationMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
@@ -47,7 +47,7 @@
      *
      * @return StopMessageLiveLocationMethod
      */
-    public static function createInline(string $inlineMessageId, array $data = null): StopMessageLiveLocationMethod
+    public static function createInline(string $inlineMessageId, ?array $data = null): StopMessageLiveLocationMethod
     {
         $instance = new static();
         $instance->inlineMessageId = $inlineMessageId;
diff -ruN a/src/Method/StopPollMethod.php b/src/Method/StopPollMethod.php
--- a/src/Method/StopPollMethod.php	2026-05-02 21:04:04.970571754 +0300
+++ b/src/Method/StopPollMethod.php	2026-05-02 21:04:14.637614646 +0300
@@ -38,7 +38,7 @@
      *
      * @return StopPollMethod
      */
-    public static function create(string $chatId, int $messageId, array $data = null): self
+    public static function create(string $chatId, int $messageId, ?array $data = null): self
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/UnbanChatMemberMethod.php b/src/Method/UnbanChatMemberMethod.php
--- a/src/Method/UnbanChatMemberMethod.php	2026-05-02 21:04:04.967219450 +0300
+++ b/src/Method/UnbanChatMemberMethod.php	2026-05-02 21:04:14.633861226 +0300
@@ -32,7 +32,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, int $userId, array $data = null): UnbanChatMemberMethod
+    public static function create($chatId, int $userId, ?array $data = null): UnbanChatMemberMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Method/UnpinChatMessageMethod.php b/src/Method/UnpinChatMessageMethod.php
--- a/src/Method/UnpinChatMessageMethod.php	2026-05-02 21:04:04.978264175 +0300
+++ b/src/Method/UnpinChatMessageMethod.php	2026-05-02 21:04:14.647217778 +0300
@@ -31,7 +31,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($chatId, array $data = null): UnpinChatMessageMethod
+    public static function create($chatId, ?array $data = null): UnpinChatMessageMethod
     {
         $instance = new static();
         $instance->chatId = $chatId;
diff -ruN a/src/Traits/AliasMethodTrait.php b/src/Traits/AliasMethodTrait.php
--- a/src/Traits/AliasMethodTrait.php	2026-05-02 21:04:04.926410225 +0300
+++ b/src/Traits/AliasMethodTrait.php	2026-05-02 21:04:14.585105752 +0300
@@ -39,7 +39,7 @@
      *
      * @return mixed
      */
-    abstract public function call(MethodInterface $method, string $type = null);
+    abstract public function call(MethodInterface $method, ?string $type = null);
 
     /**
      * @param AddMethodAliasInterface $method
diff -ruN a/src/Traits/GetMethodTrait.php b/src/Traits/GetMethodTrait.php
--- a/src/Traits/GetMethodTrait.php	2026-05-02 21:04:04.928387976 +0300
+++ b/src/Traits/GetMethodTrait.php	2026-05-02 21:04:14.588234436 +0300
@@ -40,7 +40,7 @@
      *
      * @return mixed
      */
-    abstract public function call(MethodInterface $method, string $type = null);
+    abstract public function call(MethodInterface $method, ?string $type = null);
 
     /**
      * @throws ResponseException
diff -ruN a/src/Type/ChatPermissionsType.php b/src/Type/ChatPermissionsType.php
--- a/src/Type/ChatPermissionsType.php	2026-05-02 21:04:04.933824861 +0300
+++ b/src/Type/ChatPermissionsType.php	2026-05-02 21:04:14.594577428 +0300
@@ -80,7 +80,7 @@
      *
      * @return ChatPermissionsType
      */
-    public static function create(array $data = null): ChatPermissionsType
+    public static function create(?array $data = null): ChatPermissionsType
     {
         $instance = new static();
         if ($data) {
diff -ruN a/src/Type/InlineKeyboardButtonType.php b/src/Type/InlineKeyboardButtonType.php
--- a/src/Type/InlineKeyboardButtonType.php	2026-05-02 21:04:04.934765383 +0300
+++ b/src/Type/InlineKeyboardButtonType.php	2026-05-02 21:04:14.596017939 +0300
@@ -103,7 +103,7 @@
      *
      * @return InlineKeyboardButtonType
      */
-    public static function create(string $text, array $data = null): InlineKeyboardButtonType
+    public static function create(string $text, ?array $data = null): InlineKeyboardButtonType
     {
         $instance = new static();
         $instance->text = $text;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultArticleType.php b/src/Type/InlineQueryResult/InlineQueryResultArticleType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultArticleType.php	2026-05-02 21:04:04.946851587 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultArticleType.php	2026-05-02 21:04:14.609385866 +0300
@@ -85,7 +85,7 @@
         string $id,
         string $title,
         InputMessageContentType $inputMessageContent,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultArticleType {
         $instance = new static();
         $instance->type = self::TYPE_ARTICLE;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultAudioType.php b/src/Type/InlineQueryResult/InlineQueryResultAudioType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultAudioType.php	2026-05-02 21:04:04.950759963 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultAudioType.php	2026-05-02 21:04:14.614234722 +0300
@@ -76,7 +76,7 @@
         string $id,
         string $audioUrl,
         string $title,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultAudioType {
         $instance = new static();
         $instance->type = self::TYPE_AUDIO;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedAudioType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedAudioType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedAudioType.php	2026-05-02 21:04:04.950978667 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedAudioType.php	2026-05-02 21:04:14.614572423 +0300
@@ -51,7 +51,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(string $id, string $audioFileId, array $data = null): InlineQueryResultCachedAudioType
+    public static function create(string $id, string $audioFileId, ?array $data = null): InlineQueryResultCachedAudioType
     {
         $instance = new static();
         $instance->type = self::TYPE_AUDIO;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedDocumentType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedDocumentType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedDocumentType.php	2026-05-02 21:04:04.950552426 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedDocumentType.php	2026-05-02 21:04:14.613898854 +0300
@@ -68,7 +68,7 @@
         string $id,
         string $title,
         string $documentFileId,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultCachedDocumentType {
         $instance = new static();
         $instance->type = static::TYPE_DOCUMENT;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedGifType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedGifType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedGifType.php	2026-05-02 21:04:04.948078061 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedGifType.php	2026-05-02 21:04:14.610885501 +0300
@@ -62,7 +62,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(string $id, string $gifFileId, array $data = null): InlineQueryResultCachedGifType
+    public static function create(string $id, string $gifFileId, ?array $data = null): InlineQueryResultCachedGifType
     {
         $instance = new static();
         $instance->type = static::TYPE_GIF;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedMpeg4GifType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedMpeg4GifType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedMpeg4GifType.php	2026-05-02 21:04:04.946456137 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedMpeg4GifType.php	2026-05-02 21:04:14.608706422 +0300
@@ -61,7 +61,7 @@
     public static function create(
         string $id,
         string $mpeg4FileId,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultCachedMpeg4GifType {
         $instance = new static();
         $instance->type = static::TYPE_MPEG4GIF;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedPhotoType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedPhotoType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedPhotoType.php	2026-05-02 21:04:04.947257287 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedPhotoType.php	2026-05-02 21:04:14.609971895 +0300
@@ -64,7 +64,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(string $id, string $photoFileId, array $data = null): InlineQueryResultCachedPhotoType
+    public static function create(string $id, string $photoFileId, ?array $data = null): InlineQueryResultCachedPhotoType
     {
         $instance = new static();
         $instance->type = static::TYPE_PHOTO;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedStickerType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedStickerType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedStickerType.php	2026-05-02 21:04:04.950352221 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedStickerType.php	2026-05-02 21:04:14.613570444 +0300
@@ -42,7 +42,7 @@
     public static function create(
         string $id,
         string $stickerFileId,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultCachedStickerType {
         $instance = new static();
         $instance->type = static::TYPE_STICKER;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedVideoType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedVideoType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedVideoType.php	2026-05-02 21:04:04.948394429 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedVideoType.php	2026-05-02 21:04:14.611181870 +0300
@@ -71,7 +71,7 @@
         string $id,
         string $videoFileId,
         string $title,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultCachedVideoType {
         $instance = new static();
         $instance->type = static::TYPE_VIDEO;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultCachedVoiceType.php b/src/Type/InlineQueryResult/InlineQueryResultCachedVoiceType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultCachedVoiceType.php	2026-05-02 21:04:04.949003666 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultCachedVoiceType.php	2026-05-02 21:04:14.612113184 +0300
@@ -62,7 +62,7 @@
         string $id,
         string $voiceFileId,
         string $title,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultCachedVoiceType {
         $instance = new static();
         $instance->type = static::TYPE_VOICE;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultContactType.php b/src/Type/InlineQueryResult/InlineQueryResultContactType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultContactType.php	2026-05-02 21:04:04.947751859 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultContactType.php	2026-05-02 21:04:14.610591174 +0300
@@ -89,7 +89,7 @@
         string $id,
         string $phoneNumber,
         string $firstName,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultContactType {
         $instance = new static();
         $instance->type = static::TYPE_CONTACT;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultDocumentType.php b/src/Type/InlineQueryResult/InlineQueryResultDocumentType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultDocumentType.php	2026-05-02 21:04:04.950108935 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultDocumentType.php	2026-05-02 21:04:14.613285117 +0300
@@ -107,7 +107,7 @@
         string $title,
         string $documentUrl,
         string $mimeType,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultDocumentType {
         $instance = new static();
         $instance->type = static::TYPE_DOCUMENT;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultGameType.php b/src/Type/InlineQueryResult/InlineQueryResultGameType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultGameType.php	2026-05-02 21:04:04.946254849 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultGameType.php	2026-05-02 21:04:14.608350679 +0300
@@ -31,7 +31,7 @@
      *
      * @return InlineQueryResultGameType
      */
-    public static function create(string $id, string $gameShortName, array $data = null): InlineQueryResultGameType
+    public static function create(string $id, string $gameShortName, ?array $data = null): InlineQueryResultGameType
     {
         $instance = new static();
         $instance->type = static::TYPE_GAME;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultGifType.php b/src/Type/InlineQueryResult/InlineQueryResultGifType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultGifType.php	2026-05-02 21:04:04.949423741 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultGifType.php	2026-05-02 21:04:14.612667880 +0300
@@ -90,7 +90,7 @@
         string $id,
         string $gifUrl,
         string $thumbUrl,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultGifType {
         $instance = new static();
         $instance->type = static::TYPE_GIF;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultLocationType.php b/src/Type/InlineQueryResult/InlineQueryResultLocationType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultLocationType.php	2026-05-02 21:04:04.948802129 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultLocationType.php	2026-05-02 21:04:14.611812606 +0300
@@ -107,7 +107,7 @@
         float $latitude,
         float $longitude,
         string $title,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultLocationType {
         $instance = new static();
         $instance->type = static::TYPE_LOCATION;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultMpeg4GifType.php b/src/Type/InlineQueryResult/InlineQueryResultMpeg4GifType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultMpeg4GifType.php	2026-05-02 21:04:04.947046999 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultMpeg4GifType.php	2026-05-02 21:04:14.609687610 +0300
@@ -90,7 +90,7 @@
         string $id,
         string $mpeg4Url,
         string $thumbUrl,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultMpeg4GifType {
         $instance = new static();
         $instance->type = static::TYPE_MPEG4GIF;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultPhotoType.php b/src/Type/InlineQueryResult/InlineQueryResultPhotoType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultPhotoType.php	2026-05-02 21:04:04.947552655 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultPhotoType.php	2026-05-02 21:04:14.610296972 +0300
@@ -90,7 +90,7 @@
         string $id,
         string $photoUrl,
         string $thumbUrl,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultPhotoType {
         $instance = new static();
         $instance->type = static::TYPE_PHOTO;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultVenueType.php b/src/Type/InlineQueryResult/InlineQueryResultVenueType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultVenueType.php	2026-05-02 21:04:04.946647133 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultVenueType.php	2026-05-02 21:04:14.609098747 +0300
@@ -101,7 +101,7 @@
         float $longitude,
         string $title,
         string $address,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultVenueType {
         $instance = new static();
         $instance->type = static::TYPE_VENUE;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultVideoType.php b/src/Type/InlineQueryResult/InlineQueryResultVideoType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultVideoType.php	2026-05-02 21:04:04.948596342 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultVideoType.php	2026-05-02 21:04:14.611494030 +0300
@@ -117,7 +117,7 @@
         string $mimeType,
         string $thumbUrl,
         string $title,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultVideoType {
         $instance = new static();
         $instance->type = static::TYPE_VIDEO;
diff -ruN a/src/Type/InlineQueryResult/InlineQueryResultVoiceType.php b/src/Type/InlineQueryResult/InlineQueryResultVoiceType.php
--- a/src/Type/InlineQueryResult/InlineQueryResultVoiceType.php	2026-05-02 21:04:04.949196704 +0300
+++ b/src/Type/InlineQueryResult/InlineQueryResultVoiceType.php	2026-05-02 21:04:14.612379511 +0300
@@ -69,7 +69,7 @@
         string $id,
         string $voiceUrl,
         string $title,
-        array $data = null
+        ?array $data = null
     ): InlineQueryResultVoiceType {
         $instance = new static();
         $instance->type = static::TYPE_VOICE;
diff -ruN a/src/Type/InputMedia/InputMediaAnimationType.php b/src/Type/InputMedia/InputMediaAnimationType.php
--- a/src/Type/InputMedia/InputMediaAnimationType.php	2026-05-02 21:04:04.939461576 +0300
+++ b/src/Type/InputMedia/InputMediaAnimationType.php	2026-05-02 21:04:14.600687799 +0300
@@ -56,7 +56,7 @@
      *
      * @return InputMediaAnimationType
      */
-    public static function create($media, array $data = null): InputMediaAnimationType
+    public static function create($media, ?array $data = null): InputMediaAnimationType
     {
         $instance = new static();
         $instance->media = $media;
diff -ruN a/src/Type/InputMedia/InputMediaAudioType.php b/src/Type/InputMedia/InputMediaAudioType.php
--- a/src/Type/InputMedia/InputMediaAudioType.php	2026-05-02 21:04:04.939113000 +0300
+++ b/src/Type/InputMedia/InputMediaAudioType.php	2026-05-02 21:04:14.600440388 +0300
@@ -57,7 +57,7 @@
      *
      * @return InputMediaAudioType
      */
-    public static function create($media, array $data = null): InputMediaAudioType
+    public static function create($media, ?array $data = null): InputMediaAudioType
     {
         $instance = new static();
         $instance->media = $media;
diff -ruN a/src/Type/InputMedia/InputMediaDocumentType.php b/src/Type/InputMedia/InputMediaDocumentType.php
--- a/src/Type/InputMedia/InputMediaDocumentType.php	2026-05-02 21:04:04.939989731 +0300
+++ b/src/Type/InputMedia/InputMediaDocumentType.php	2026-05-02 21:04:14.601176914 +0300
@@ -41,7 +41,7 @@
      *
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create($media, array $data = null): InputMediaDocumentType
+    public static function create($media, ?array $data = null): InputMediaDocumentType
     {
         $instance = new static();
         $instance->media = $media;
diff -ruN a/src/Type/InputMedia/InputMediaPhotoType.php b/src/Type/InputMedia/InputMediaPhotoType.php
--- a/src/Type/InputMedia/InputMediaPhotoType.php	2026-05-02 21:04:04.938913462 +0300
+++ b/src/Type/InputMedia/InputMediaPhotoType.php	2026-05-02 21:04:14.600172435 +0300
@@ -24,7 +24,7 @@
      *
      * @return InputMediaPhotoType
      */
-    public static function create($media, array $data = null): InputMediaPhotoType
+    public static function create($media, ?array $data = null): InputMediaPhotoType
     {
         $instance = new static();
         $instance->media = $media;
diff -ruN a/src/Type/InputMedia/InputMediaVideoType.php b/src/Type/InputMedia/InputMediaVideoType.php
--- a/src/Type/InputMedia/InputMediaVideoType.php	2026-05-02 21:04:04.938711425 +0300
+++ b/src/Type/InputMedia/InputMediaVideoType.php	2026-05-02 21:04:14.599929898 +0300
@@ -63,7 +63,7 @@
      *
      * @return InputMediaVideoType
      */
-    public static function create($media, array $data = null): InputMediaVideoType
+    public static function create($media, ?array $data = null): InputMediaVideoType
     {
         $instance = new static();
         $instance->media = $media;
diff -ruN a/src/Type/InputMessageContent/InputContactMessageContentType.php b/src/Type/InputMessageContent/InputContactMessageContentType.php
--- a/src/Type/InputMessageContent/InputContactMessageContentType.php	2026-05-02 21:04:04.954286847 +0300
+++ b/src/Type/InputMessageContent/InputContactMessageContentType.php	2026-05-02 21:04:14.618481966 +0300
@@ -54,7 +54,7 @@
     public static function create(
         string $phoneNumber,
         string $firstName,
-        array $data = null
+        ?array $data = null
     ): InputContactMessageContentType {
         $instance = new static();
         $instance->phoneNumber = $phoneNumber;
diff -ruN a/src/Type/InputMessageContent/InputLocationMessageContentType.php b/src/Type/InputMessageContent/InputLocationMessageContentType.php
--- a/src/Type/InputMessageContent/InputLocationMessageContentType.php	2026-05-02 21:04:04.955292409 +0300
+++ b/src/Type/InputMessageContent/InputLocationMessageContentType.php	2026-05-02 21:04:14.619453154 +0300
@@ -66,7 +66,7 @@
     public static function create(
         float $latitude,
         float $longitude,
-        array $data = null
+        ?array $data = null
     ): InputLocationMessageContentType {
         $instance = new static();
         $instance->latitude = $latitude;
diff -ruN a/src/Type/InputMessageContent/InputTextMessageContentType.php b/src/Type/InputMessageContent/InputTextMessageContentType.php
--- a/src/Type/InputMessageContent/InputTextMessageContentType.php	2026-05-02 21:04:04.954040477 +0300
+++ b/src/Type/InputMessageContent/InputTextMessageContentType.php	2026-05-02 21:04:14.618147473 +0300
@@ -43,7 +43,7 @@
     /**
      * @throws \TgBotApi\BotApiBase\Exception\BadArgumentException
      */
-    public static function create(string $messageText, array $data = null): InputTextMessageContentType
+    public static function create(string $messageText, ?array $data = null): InputTextMessageContentType
     {
         $instance = new static();
         $instance->messageText = $messageText;
diff -ruN a/src/Type/InputMessageContent/InputVenueMessageContent.php b/src/Type/InputMessageContent/InputVenueMessageContent.php
--- a/src/Type/InputMessageContent/InputVenueMessageContent.php	2026-05-02 21:04:04.954685630 +0300
+++ b/src/Type/InputMessageContent/InputVenueMessageContent.php	2026-05-02 21:04:14.618833834 +0300
@@ -68,7 +68,7 @@
         float $longitude,
         string $title,
         string $address,
-        array $data = null
+        ?array $data = null
     ): InputVenueMessageContent {
         $instance = new static();
         $instance->latitude = $latitude;
diff -ruN a/src/Type/KeyboardButtonType.php b/src/Type/KeyboardButtonType.php
--- a/src/Type/KeyboardButtonType.php	2026-05-02 21:04:04.951289910 +0300
+++ b/src/Type/KeyboardButtonType.php	2026-05-02 21:04:14.615025164 +0300
@@ -62,7 +62,7 @@
     /**
      * @throws BadArgumentException
      */
-    public static function create(string $text, array $data = null): KeyboardButtonType
+    public static function create(string $text, ?array $data = null): KeyboardButtonType
     {
         $instance = new static();
         $instance->text = $text;
diff -ruN a/src/Type/LoginUrlType.php b/src/Type/LoginUrlType.php
--- a/src/Type/LoginUrlType.php	2026-05-02 21:04:04.943042542 +0300
+++ b/src/Type/LoginUrlType.php	2026-05-02 21:04:14.604955543 +0300
@@ -63,7 +63,7 @@
      *
      * @return LoginUrlType
      */
-    public static function create(string $url, array $data = null): LoginUrlType
+    public static function create(string $url, ?array $data = null): LoginUrlType
     {
         $instance = new static();
         $instance->url = $url;
diff -ruN a/src/Type/MenuButtonType.php b/src/Type/MenuButtonType.php
--- a/src/Type/MenuButtonType.php	2026-05-02 21:04:04.956633714 +0300
+++ b/src/Type/MenuButtonType.php	2026-05-02 21:04:14.620778293 +0300
@@ -49,7 +49,7 @@
      *
      * @return MenuButtonType
      */
-    public static function create(string $type, array $data = null): MenuButtonType
+    public static function create(string $type, ?array $data = null): MenuButtonType
     {
         $instance = new static();
         $instance->type = $type;
diff -ruN a/src/Type/MessageEntityType.php b/src/Type/MessageEntityType.php
--- a/src/Type/MessageEntityType.php	2026-05-02 21:04:04.940598093 +0300
+++ b/src/Type/MessageEntityType.php	2026-05-02 21:04:14.601968189 +0300
@@ -77,7 +77,7 @@
      */
     public $language;
 
-    public static function create(string $type, int $offset, int $length, array $data = null): MessageEntityType
+    public static function create(string $type, int $offset, int $length, ?array $data = null): MessageEntityType
     {
         $instance = new static();
         $instance->type = $type;
diff -ruN a/src/Type/ReplyKeyboardMarkupType.php b/src/Type/ReplyKeyboardMarkupType.php
--- a/src/Type/ReplyKeyboardMarkupType.php	2026-05-02 21:04:04.951599320 +0300
+++ b/src/Type/ReplyKeyboardMarkupType.php	2026-05-02 21:04:14.615352074 +0300
@@ -61,7 +61,7 @@
      *
      * @return ReplyKeyboardMarkupType
      */
-    public static function create(array $keyboard, array $data = null): ReplyKeyboardMarkupType
+    public static function create(array $keyboard, ?array $data = null): ReplyKeyboardMarkupType
     {
         $instance = new static();
         $instance->keyboard = $keyboard;
